diff -Naur src/file.c src/file.c
--- src/file.c	2011-03-25 08:18:24 +0000
+++ src/file.c	2013-03-14 16:29:12 +0000
@@ -1808,7 +1808,7 @@
  *  it returns NULL and stores an error code in errno. An attempt to read a 
  *  normal file in packed mode will cause errno to be set to EDOM.
  */
-PACKFILE *pack_fopen(AL_CONST char *filename, AL_CONST char *mode)
+PACKFILE *__old_pack_fopen(AL_CONST char *filename, AL_CONST char *mode)
 {
    char tmp[1024];
    int fd;
diff -Naur src/win/wkeybd.c src/win/wkeybd.c
--- src/win/wkeybd.c	2010-07-24 22:36:22 +0000
+++ src/win/wkeybd.c	2013-03-14 16:57:44 +0000
@@ -40,12 +40,12 @@
 static HANDLE key_input_event = NULL;
 static HANDLE key_input_processed_event = NULL;
 static LPDIRECTINPUT key_dinput = NULL;
-static LPDIRECTINPUTDEVICE key_dinput_device = NULL;
+LPDIRECTINPUTDEVICE key_dinput_device = NULL;
 
 
 /* lookup table for converting DIK_* scancodes into Allegro KEY_* codes */
 /* this table was from pckeys.c  */
-static const unsigned char hw_to_mycode[256] = {
+const unsigned char hw_to_mycode[256] = {
    /* 0x00 */ 0, KEY_ESC, KEY_1, KEY_2,
    /* 0x04 */ KEY_3, KEY_4, KEY_5, KEY_6,
    /* 0x08 */ KEY_7, KEY_8, KEY_9, KEY_0,
@@ -322,11 +322,26 @@
 {
    HWND allegro_wnd = win_get_window();
    static int ignore_three_finger_flag = FALSE;
+   static BOOL just_pressed_alt = FALSE;
+
+   if (scancode == DIK_LMENU)
+   {
+     just_pressed_alt = TRUE;
+   }
+   else if (scancode != DIK_TAB)
+   {
+     just_pressed_alt = FALSE;
+   }
+
    /* ignore special Windows keys (alt+tab, alt+space, (ctrl|alt)+esc) */
    if (((scancode == DIK_TAB) && (_key_shifts & KB_ALT_FLAG))
        || ((scancode == DIK_SPACE) && (_key_shifts & KB_ALT_FLAG))
-       || ((scancode == DIK_ESCAPE) && (_key_shifts & (KB_CTRL_FLAG | KB_ALT_FLAG))))
+       || ((scancode == DIK_ESCAPE) && (_key_shifts & (KB_CTRL_FLAG | KB_ALT_FLAG)))
+       || ((scancode == DIK_TAB) && (just_pressed_alt)))
+   {
+      just_pressed_alt = FALSE;
       return;
+   }
 
    /* alt+F4 triggers a WM_CLOSE under Windows */
    if ((scancode == DIK_F4) && (_key_shifts & KB_ALT_FLAG)) {
diff -Naur src/win/wmouse.c src/win/wmouse.c
--- src/win/wmouse.c	2010-07-24 22:36:22 +0000
+++ src/win/wmouse.c	2013-03-14 16:30:05 +0000
@@ -82,7 +82,7 @@
 #define DINPUT_BUFFERSIZE 256
 static HANDLE mouse_input_event = NULL;
 static LPDIRECTINPUT mouse_dinput = NULL;
-static LPDIRECTINPUTDEVICE mouse_dinput_device = NULL;
+LPDIRECTINPUTDEVICE mouse_dinput_device = NULL;
 
 static int dinput_buttons = 0;
 static int dinput_wheel = FALSE;
diff -Naur src/win/wwnd.c src/win/wwnd.c
--- src/win/wwnd.c	2008-10-17 02:27:30 +0000
+++ src/win/wwnd.c	2013-03-14 16:29:01 +0000
@@ -36,7 +36,7 @@
 
 
 /* general */
-static HWND allegro_wnd = NULL;
+HWND allegro_wnd = NULL;
 char wnd_title[WND_TITLE_SIZE];  /* ASCII string */
 int wnd_x = 0;
 int wnd_y = 0;
--- src/win/wdsound.c	2009-12-16 13:52:54 +0000
+++ src/win/wdsound.c	2013-03-14 17:28:55 +0000
@@ -151,7 +151,7 @@
 
 
 /* sound driver globals */
-static LPDIRECTSOUND directsound = NULL;
+LPDIRECTSOUND directsound = NULL;
 static LPDIRECTSOUNDBUFFER prim_buf = NULL;
 static long int initial_volume;
 static int _freq, _bits, _stereo;
